File: commands/start.py
"""Handlers for the `/start` command."""

from __future__ import annotations

import logging

from aiogram import Router
from aiogram.enums.parse_mode import ParseMode
from aiogram.filters import CommandStart
from aiogram.types import Message

from functions.mongodb import save_user

router = Router()

logger = logging.getLogger(__name__)


@router.message(CommandStart())
async def handle_start(message: Message) -> None:
    """Send usage instructions for the bot when `/start` is invoked."""
    user = message.from_user
    if user is not None:
        try:
            await save_user(user.id, user.username, user.first_name)
        except Exception as exc:  # pragma: no cover - persistence errors shouldn't break /start
            logger.warning("Failed to persist user %s: %s", user.id, exc)

    await message.answer(
        "Send the account string in the format:\n"
        "<code>email|password|refresh_token|client_id</code>\n\n"
        "Only Hotmail/Outlook addresses are accepted.",
        parse_mode=ParseMode.HTML,
    )

File: src/commands/start.js
import { Markup } from "telegraf";
import { saveUser } from "../functions/mongodb.js";

export function registerStart(bot) {
  bot.start(async (ctx) => {
    const user = ctx.from;
    if (user) {
      try {
        await saveUser(user.id, user.username, user.first_name);
      } catch (err) {
        console.warn("Failed to persist user", user.id, err);
      }
    }

    const keyboard = Markup.inlineKeyboard([
      [Markup.button.callback("🇧🇩 বাংলা", "lang:bn"), Markup.button.callback("🇬🇧 English", "lang:en")],
    ]);

    await ctx.reply(
      "🌐 Choose your language\nPlease pick how you'd like to use the bot.",
      keyboard
    );
  });

  bot.action("lang:bn", async (ctx) => {
    await ctx.answerCbQuery();
    const banglaText =
      "আমাদের বটে স্বাগতম!\n" +
      "আপনি এখন খুব সহজে আপনার Hotmail বা Outlook মেইল থেকে Facebook কোড পেতে পারেন —\n" +
      "শুধু আপনার মেইল পাঠান, আমরা সঙ্গে সঙ্গে কোডটা দিয়ে দেব।\n\n" +
      "🔐 অ্যাকাউন্ট তথ্য পাঠাতে এই ফরম্যাট ব্যবহার করুন:\n" +
      "<code>email|password|refresh_token|client_id</code>\n\n" +
      "শুধুমাত্র Hotmail/Outlook ঠিকানা গ্রহণ করা হবে।";
    await ctx.editMessageText(banglaText, { parse_mode: "HTML" });
  });

  bot.action("lang:en", async (ctx) => {
    await ctx.answerCbQuery();
    const englishText =
      "Welcome to our bot!\n" +
      "You can get your Hotmail or Outlook Facebook codes instantly —\n" +
      "just send your email, and we’ll fetch the code for you.\n\n" +
      "🔐 Send the account string in this format:\n" +
      "<code>email|password|refresh_token|client_id</code>\n\n" +
      "Only Hotmail/Outlook addresses are accepted.";
    await ctx.editMessageText(englishText, { parse_mode: "HTML" });
  });
}
